### ISSUE TITLE
Source Maps Debugger Modal shows warning even when event is symbolicated correctly

### ISSUE THREAD

**Issue by ShaneMcNamara** [getsentry/sentry-react-native#3667]
**OS:**
- [ ]  Windows
- [x]  MacOS
- [ ]  Linux

**Platform:**
- [x]  iOS
- [ ]  Android

**SDK:**
- [x]  `@sentry/react-native` (>= 1.0.0)
- [ ]  `react-native-sentry` (<= 0.43.2)

**SDK version:**  "@sentry/react-native": "5.10.0",

**`react-native` version:**    "react-native": "0.72.6",

**Are you using Expo?**
- [x]  Yes
- [ ]  No

**Are you using sentry.io or on-premise?**
- [x]  sentry.io (SaaS)
- [ ]  on-premise

**If you are using sentry.io, please post a link to your issue so we can take a look:**

[[Link to issue]](https://ontraccr.sentry.io/issues/5046145973/events/2cced7dbf7c2450e848b9e50c08cf334/?project=5891694&query=release%3Amobile%403.25.0&referrer=next-event)

**Configuration:**

(`@sentry/react-native`)
```
Sentry.init({
  dsn: config.sentryDSN,
  environment: config.env,
  enableInExpoDevelopment: config.enableInExpoDevelopment,
  release: `mobile@${version}`,
  dist: commit,
  integrations: [
    new RewriteFrames({
      iteratee: frame => {
        if (frame.filename) {
          // the values depend on what names you give the bundle files you are uploading to Sentry
          frame.filename =
            Platform.OS === 'android' ? 'app:///index.android.bundle' : 'app:///main.jsbundle';
        }
        return frame;
      },
    }),
  ],
  debug: config.env === 'local'
});
```

---
**I have the following issue:**

We receive a warning that the stack frame path does not match a source file artifact name, and our stack trace is not useful.

<img width="418" alt="Screenshot 2024-03-07 at 4 29 07 PM" src="https://github.com/getsentry/sentry-react-native/assets/9358722/099e8c20-97f6-4be0-9cea-784933ce7308">

yet the artifact is there

<img width="895" alt="Screenshot 2024-03-07 at 4 29 52 PM" src="https://github.com/getsentry/sentry-react-native/assets/9358722/8bef03a9-dcdc-4f66-b32e-0ecbd1e0406c">

There was a similar issue reported here: https://github.com/getsentry/sentry-react-native/issues/3492 but in that instance they were not setting dist, we however are setting dist. I'm not sure if the fact that the minified files are 0 bytes matter or not.

**Comment by krystofwoldrich** [comment]
Hi @ShaneMcNamara,
thank you for the message,
you are correct the `dist` is correct, but the bundle `~/main.jsbundle` doesn't reference the `~/ios-df3a612bfd09bd56f8d2a7349f9cdfac.map` source map file. 

The referencing is done during the upload of the source maps, can you share how you upload the source maps and output of that command?

The reference is displayed in Sentry like this:
![Screenshot 2024-03-12 at 13 03 34](https://github.com/getsentry/sentry-react-native/assets/31292499/95a7fdb6-f595-4262-a49b-4ac265cd9dc0)

PS: Since https://github.com/getsentry/sentry-react-native/releases/tag/5.16.0 and Expo SDK 50 we offer a new way of uploading source maps.

**Comment by ShaneMcNamara** [comment]
Hi @krystofwoldrich,

Thanks for the response

We've essentially followed the guide here: https://docs.expo.dev/guides/using-sentry/#uploading-source-maps-for-updates for SDK 49 and below.

In the deploy step of our pipeline we run these commands:

```
 - eas update --channel $CHANNEL --message=$BITBUCKET_COMMIT
 # https://docs.expo.dev/guides/using-sentry/
- mv dist/bundles/android-*.hbc dist/bundles/index.android.bundle
- mv dist/bundles/ios-*.hbc dist/bundles/main.jsbundle
- node_modules/@sentry/cli/bin/sentry-cli releases files mobile@$RELEASE upload-sourcemaps --dist $BITBUCKET_COMMIT --rewrite dist/bundles/*  \;
```

Few comments/questions:

1. We have modified the `dist` to use the commit hash instead of he `updateId` generated from expo, is that an issue?
2. The move commands are to fulfill this instruction in the expo guide "Copy or rename the bundle names in the dist/bundles folder to match index.android.bundle (Android) or main.jsbundle (iOS)." Do we need to also rename the map files?
3. We condensed the upload command to one (instead of doing it for each platform) is this an issue?

We are targeting to upgrade to Expo SDK 50 in a month, but I was hoping to get source maps up and running before upgrading. Obviously upgrading SDK versions has a much larger risk/regression scope.

**Comment by krystofwoldrich** [comment]
@ShaneMcNamara Thank you for the details, yes, you have to upload the Android and iOS separately, so the `sentry-cli` can correctly link the bundle and source map.

Let us know if that helped.

**Comment by ShaneMcNamara** [comment]
@krystofwoldrich  - It doesn't seem to have helped. 

Here are the new update steps and outputs:
```
- find dist/bundles/android-*.map -exec node_modules/@sentry/cli/bin/sentry-cli releases files mobile@$RELEASE upload-sourcemaps --dist $BITBUCKET_COMMIT-android --rewrite dist/bundles/index.android.bundle {} --org ontraccr \;
> Found 1 file
> Found 1 file
> Analyzing 2 sources
> Rewriting sources
> Adding source map references
> Bundled 2 files for upload
> Bundle ID: ce9752fe-0630-52cc-89fd-b42d2950e79f
> Uploaded files to Sentry
> File upload complete (processing pending on server)
> Organization: ontraccr
> Project: mobile
> Release: mobile@3.25.0
> Dist: 6541d0b69ff81f902d51d46f8fa1f54287bc5254-android
> Upload type: artifact bundle
Source Map Upload Report
  Minified Scripts
    ~/index.android.bundle (sourcemap at android-757cbaadc11f4c562d62754b597392ca.map)
  Source Maps
    ~/android-757cbaadc11f4c562d62754b597392ca.map
```

```
- find dist/bundles/ios-*.map -exec node_modules/@sentry/cli/bin/sentry-cli releases files mobile@$RELEASE upload-sourcemaps --dist $BITBUCKET_COMMIT-ios --rewrite dist/bundles/main.jsbundle {} --org ontraccr \;
> Found 1 file
> Found 1 file
> Analyzing 2 sources
> Rewriting sources
> Adding source map references
> Bundled 2 files for upload
> Bundle ID: eb6ff50e-0f67-5187-aef0-b13991295998
> Uploaded files to Sentry
> File upload complete (processing pending on server)
> Organization: ontraccr
> Project: mobile
> Release: mobile@3.25.0
> Dist: 6541d0b69ff81f902d51d46f8fa1f54287bc5254-ios
> Upload type: artifact bundle
Source Map Upload Report
  Minified Scripts
    ~/main.jsbundle (sourcemap at ios-e7b2ed22b8be77d1ce047ac551f123f0.map)
  Source Maps
    ~/ios-e7b2ed22b8be77d1ce047ac551f123f0.map
```

[Issue Link](https://ontraccr.sentry.io/issues/5067611085/?environment=alpha&project=5891694&query=is%3Aunresolved&referrer=issue-stream&statsPeriod=14d&stream_index=0)

Error Message:
<img width="446" alt="Screenshot 2024-03-14 at 2 52 21 PM" src="https://github.com/getsentry/sentry-react-native/assets/9358722/e5650cb1-7720-4afa-ae4b-2a4d4843ea4b">

And associated release+dist:
<img width="842" alt="Screenshot 2024-03-14 at 2 52 28 PM" src="https://github.com/getsentry/sentry-react-native/assets/9358722/108835a8-8915-4029-95db-4cd077fcdb7c">


Please let me know if there is any other information you need

**Comment by krystofwoldrich** [comment]
@ShaneMcNamara Thank you for the link, we will check why the warning is still showing, 

but as fas, as I see your stack trace is correctly symbolicated, so you can ignore the `unminify` button.

**Comment by rikur** [comment]
+1, I have symbolicated stack traces yet the warning appears

**Comment by krystofwoldrich** [comment]
- Closing as duplicate of https://github.com/getsentry/sentry-react-native/issues/3667
